torch/csrc/jit/codegen/cuda/lower_predicate.cpp

```diff
@@ -175,6 +177,488 @@ std::vector<kir::Expr*> generateConditionalFromPredicate(
   return mutated_exprs;
 }
 
+namespace {
+
+//! Analyze whether IterDomain can be statically determined to be safe
+//! without bounds-checking predicates.
```

Comment:
This is a little vague, can you mention this analysis checks if there cannot be any out of bounds accesses based on the loop nest pattern? As opposed to detecting if we're just filling in "out of bounds" elements in a local buffer.

Reply:
That is correct. Note this is not new code. I just moved from another file.

The idea is that when an extent is evenly divided by a split, the resulting doubly nested loop won't go beyond the extent. At least, that was the original idea when I added this (I believe that was when I was working on GEMM). 

However, this doesn't seem to work when parallelized and the number of threads is larger than the extent. It should still be safe if not parallelized. I'll temporary disable this and later work on a safer version possibly merged with the new logic added in this PR.